silverstripe/silverstripe-standards
# NEW Add method annotation checks

Since this doesn't have CI yet, you'll have to check the CI in creative-commoners instead of on this PR directly.
See https://github.com/creative-commoners/silverstripe-standards/actions/workflows/ci.yml

Note that the PDO failure will be fixed by https://github.com/silverstripe/gha-ci/issues/97

## How to test this manually

1. Add this repo and `phpstan/extension-installer` as dev dependencies
2. Add a `phpstan.neon` or `phpstan.neon.dist` file to your project root with this content:
    ```neon
    parameters:
      paths:
        - vendor/silverstripe/framework/src/Security
    ```
3. Run `vendor/bin/phpstan`
    This will run it against the path set above, and due to the rules set in this repo it will only run our custom rule.
4. It should be pretty quick, and the only failures you should see should relate to `@method` annotations.

## Issue
- https://github.com/silverstripe/.github/issues/171
